### PR TITLE
[Pagination] adding loading support to Pagination component

### LINKED ISSUE #2316: Add loading indicator to Pagination
## Feature request summary

When paging a remote date source we often want to also render an additional loading indicator while the next or previous page is being fetched. This would be best supported directly by the `Pagination` component with a new `loading` prop.

## Rationale

This would eliminate some custom external implementation to swap out the `Pagination` component while loading, by allowing the `Pagination` component to fully provide the loading indicator in a reasonable layout.

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by patsissons**
### WHY are these changes introduced?

Fixes #2316 

### WHAT is this pull request doing?

Pagination will render a `Spinner` when `loading` is `true`

Kapture 2019-10-17 at 9 52 23

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```tsx
import React, {useState} from 'react';
import {Card, Page, Pagination, TextContainer} from '../src';

function Demo({plain}: {plain?: boolean}) {
  const [isLoading, setIsLoading] = useState(false);

  return (
    <Card sectioned title={plain ? 'Plain' : 'Button'}>
      <div style={{textAlign: 'center'}}>
        <TextContainer>
          <p>Spicy jalapeno bacon ipsum</p>
          <Pagination
            hasNext
            hasPrevious
            onNext={page}
            onPrevious={page}
            loading={isLoading}
            plain={plain}
          />
        </TextContainer>
      </div>
    </Card>
  );

  function page() {
    setIsLoading(true);

    setTimeout(() => setIsLoading(false), 2000);
  }
}

export function Playground() {
  return (
    <Page title="">
      <Demo />
      <Demo plain />
    </Page>
  );
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/components/Pagination/Pagination.scss [hunk 1]
```diff
@@ -27,6 +27,15 @@ $stacking-order: (
       }
     }
   }
+
+  &.Loading {
```
Comment (dleroux) [inline src/components/Pagination/Pagination.scss]:
```suggestion
  &.loading {
```

**Comment by dleroux** [comment]
@sarahill thoughts on this?

@dpersing should focus go back to the button that was pressed in this case?

**Comment by kyledurand** [comment]
Hey Pat, sorry to see this got closed because I think it's a solid addition (and it would be great if we didn't have to depend on the polaris-next version). I re-opened it, added a test, changelog entry, new loading label, and made a change to your css that should pass the linter. Hope that helps get this pushed through.

@dpersing could you take a look at Dan's question above: https://github.com/Shopify/polaris-react/pull/2308#issuecomment-544999199

**Review by kyledurand — APPROVED** [review]

**Comment by alex-page** [comment]
@patsissons this looks great, it's quite out of date. Do you still intend to ship it? I am going to close this PR but feel free to reopen it. 

We are trying to reduce the need to show spinners all the time. I wonder if there is another UX pattern that could help in this situation.

**Pull request closed without merging**